x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/accordion_waterfall.tsx

```diff
@@ -0,0 +1,177 @@
+ */
+import { EuiAccordion, EuiAccordionProps } from '@elastic/eui';
+import { Location } from 'history';
+import { isEmpty } from 'lodash';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Margins } from '../../../../../shared/charts/Timeline';
+import { WaterfallItem } from './WaterfallItem';
+import {
+  IWaterfall,
+  IWaterfallItem,
+} from './waterfall_helpers/waterfall_helpers';
+
+interface AccordionWaterfallProps {
+  isOpen: boolean;
+  item: IWaterfallItem;
+  level: number;
+  serviceColors: IWaterfall['serviceColors'];
+  duration: IWaterfall['duration'];
+  waterfallItemId?: string;
+  location: Location;
+  errorsPerTransaction: IWaterfall['errorsPerTransaction'];
+  childrenByParentId: Record<string, IWaterfallItem[]>;
+  onToggleEntryTransaction?: (
+    nextState: EuiAccordionProps['forceState']
+  ) => void;
+  timelineMargins: Margins;
+  onClickWaterfallItem: (item: IWaterfallItem) => void;
+}
+
+const StyledAccordion = styled(EuiAccordion).withConfig({
+  shouldForwardProp: (prop) =>
+    !['childrenCount', 'marginLeftLevel', 'hasError'].includes(prop),
+})<
+  EuiAccordionProps & {
+    childrenCount: number;
+    marginLeftLevel: number;
+    hasError: boolean;
+  }
+>`
+  .euiAccordion {
+    border-top: 1px solid ${({ theme }) => theme.eui.euiColorLightShade};
+  }
+  .euiIEFlexWrapFix {
+    width: 100%;
+    height: 48px;
+  }
+  .euiAccordion__childWrapper {
+    transition: none;
+  }
+
+  .euiAccordion__padding--l {
+    padding-top: 0;
+    padding-bottom: 0;
+  }
+
+  .euiAccordion__iconWrapper {
+    display: flex;
+    position: relative;
+    &:after {
+      content: ${(props) => `'${props.childrenCount}'`};
+      position: absolute;
+      left: 20px;
+      top: -1px;
+      z-index: 1;
+      font-size: ${({ theme }) => theme.eui.euiFontSizeXS};
+    }
+  }
+
+  ${(props) => {
+    const borderLeft = props.hasError
+      ? `2px solid ${props.theme.eui.euiColorDanger};`
+      : `1px solid ${props.theme.eui.euiColorLightShade};`;
+    return `.button_${props.id} {
+      margin-left: ${props.marginLeftLevel}px;
+      border-left: ${borderLeft}
+      &:hover {
+        background-color: ${props.theme.eui.euiColorLightestShade};
+      }
+    }`;
+    //
+  }}
+`;
+
+const WaterfallItemContainer = styled.div`
+  position: absolute;
+  width: 100%;
+  left: 0;
+`;
+
+const getAccordionState = (isOpen: boolean): EuiAccordionProps['forceState'] =>
+  isOpen ? 'open' : 'closed';
+
+export function AccordionWaterfall({
+  isOpen,
+  item,
+  level,
+  serviceColors,
+  duration,
+  childrenByParentId,
+  waterfallItemId,
+  location,
+  errorsPerTransaction,
+  timelineMargins,
+  onClickWaterfallItem,
+}: AccordionWaterfallProps) {
+  const [state, setState] = useState(getAccordionState(isOpen));
+
+  const toggleAccordion = (isAccordionOpen: boolean) => {
+    const newState = getAccordionState(isAccordionOpen);
+    setState(newState);
+  };
```

Comment:
I think I understand now what you are trying to achieve. In that case a small re-write and a comment would make it more clear:

```suggestion
  // inherit parent `isOpen` state but allow overwriting it locally
  const [isOpen, setIsOpen] = useState(props.isOpen);
  
  // ...
  
  forceState={isOpen ? 'open' : 'closed'}
```

Reply:
Yeah, so basically what happens is that the button outside the accordion component can also control the state of it.
<img width="434" alt="Screenshot 2020-08-24 at 12 27 40" src="https://user-images.githubusercontent.com/55978943/91163618-282e5d00-e6ce-11ea-9344-304f6851f711.png">
The one highlighted as `initial state`. So I set the default state of the accordion as the value received in the prop, but `toggleAccordion` can also update its state closing or opening it.

Reply:
Makes sense. You okay with my suggestion?

Reply:
Yes, I'll change it. thanks for that!!!

Reply:
changed the suggestion you made here https://github.com/elastic/kibana/pull/75259/commits/2c496f2e1092ae3c3a4a8dd927be143cbba2815b